state/accumulator.go

```diff
@@ -412,11 +415,40 @@ func (a *Accumulator) Accumulate(txn *sqlx.Tx, userID, roomID string, prevBatch
 			if ev.NID > latestNID {
 				latestNID = ev.NID
 			}
+			// if this is a redaction, try to redact the referenced event (best effort)
+			if !ev.IsState && ev.Type == "m.room.redaction" {
+				// look for top-level redacts then content.redacts (room version 11+)
+				redactsEventID := parsedEv.Get("redacts").Str
+				if redactsEventID == "" {
+					redactsEventID = parsedEv.Get("content.redacts").Str
+				}
+				if redactsEventID != "" {
+					redactTheseEventIDs = append(redactTheseEventIDs, redactsEventID)
+				}
+			}
 			newEvents = append(newEvents, ev)
 			timelineNIDs = append(timelineNIDs, ev.NID)
 		}
 	}
 
+	// if we are going to redact things, we need the room version to know the redaction algorithm
+	// so pull it out once now.
+	var roomVersion string
+	if len(redactTheseEventIDs) > 0 {
+		createEventJSON, err := a.eventsTable.SelectCreateEvent(txn, roomID)
+		if err != nil {
+			return 0, nil, fmt.Errorf("SelectCreateEvent: %s", err)
```

Comment:
```suggestion
			return 0, nil, fmt.Errorf("SelectCreateEvent: %w", err)
```

for error propagation